README.md

```diff
@@ -1,6 +1,102 @@
-# envaridator
+# Envaridator
 
-Envaridator is a tiny environment variable validation library. It's main purpose is
-parsing, registering and validating environment variables.
+Envaridator is a really small environment variable validation library.
 
-Check `examples/` for example usage.
\ No newline at end of file
+## Install
+
+Installing the package can be achieved by:
+
+```
+yarn add envaridator
+
+npm i --save envaridator
+```
+
+## Basic usage:
+
+```
+import { Envaridator } from "envaridator";
+
+let envaridator = new Envaridator();
+
+import * as toi from "@toi/toi";
+import * as toix from "@toi/toix";
+
+process.env["CORRECT_URL"] = "https://google.com";
+const okURL = envaridator.register(
+  "CORRECT_URL",
+  toi.required().and(
+    toix.str.url({
+      protocol: "https:"
+    })
+  ),
+  "description"
+);
+
+process.env["WRONG_URL"] = "httttps://google.com";
+const badURL = envaridator.register(
+  "WRONG_URL",
+  toi.required().and(
+    toix.str.url({
+      protocol: "https:"
+    })
+  ),
+  "description for WRONG_URL"
+); // SHOULD FAIL
+
+try {
+  envaridator.validate();
+} catch (err) {
+  console.log(err.message);
+}
+```
+
+This would result in a status report:
+
+```
+The following environment variables are invalid:
```

Comment:
Are you missing the variables reported here?

Reply:
No, actually that is the "status report" that `validate()` returns if there is a environment variable that failed the validation.

In this case, `WRONG_URL` was reported that failed the validation with a reason: `Invalid protocol: https:`
valid protocol: https:
```

Reply:
Sorry, my bad. Didn't see the variable bellow.